nil pointer when checking run status for dealing with Tekton race condition

Just saw the tide pod crash with:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x17a9417]

goroutine 105 [running]:
github.com/jenkins-x/lighthouse/pkg/tide.rerunPipelineRunsWithRaceConditionFailure(0x2059de0, 0xc0003979e0, 0xc000243df4, 0x2, 0xc000847880, 0xc00085a090, 0x0)
	/workspace/source/pkg/tide/pipelinerun_monitor.go:52 +0x357
github.com/jenkins-x/lighthouse/pkg/tide.(*DefaultController).Sync(0xc000440790, 0x0, 0x0)
	/workspace/source/pkg/tide/tide.go:384 +0xc5d
main.sync(0x20758e0, 0xc000440790)
	/workspace/source/cmd/tide/main.go:197 +0x35
main.main.func1()
	/workspace/source/cmd/tide/main.go:174 +0x33
github.com/jenkins-x/lighthouse/pkg/prow/interrupts.Tick.func1(0xc000ef2140, 0xc003bec400, 0xc003bec3e0, 0xc003cac070)
	/workspace/source/pkg/prow/interrupts/interrupts.go:210 +0x73
created by github.com/jenkins-x/lighthouse/pkg/prow/interrupts.Tick
	/workspace/source/pkg/prow/interrupts/interrupts.go:196 +0xbd
```

D'oh, missed a nil check.

--- a/pkg/tide/pipelinerun_monitor.go
+++ b/pkg/tide/pipelinerun_monitor.go
@@ -49,7 +49,7 @@ func rerunPipelineRunsWithRaceConditionFailure(tektonClient tektonclient.Interfa
 	// Filter through the runs to find failed ones
 	for _, run := range runs.Items {
 		runCondition := run.Status.GetCondition(kpgapis.ConditionSucceeded)
-		if runCondition.Status == untypedcorev1.ConditionFalse {
+		if runCondition != nil && runCondition.Status == untypedcorev1.ConditionFalse {
 			// If a run has failed, check its message (or reason if the message is empty) for one of the known rerun cases
 			failDescription := runCondition.Message
 			if failDescription == "" {

